services/src/main/java/org/keycloak/services/resources/RealmsResource.java

```diff
@@ -98,9 +98,14 @@ public Object getProtocol(final @PathParam("realm") String name,
                                             final @PathParam("protocol") String protocol) {
         RealmModel realm = init(name);
 
+        LoginProtocolFactory factory = (LoginProtocolFactory)session.getKeycloakSessionFactory().getProviderFactory(LoginProtocol.class, protocol);
+        if(factory == null){
+            logger.debugv("protocol %s not found", protocol);
+            throw new NotFoundException("Protocol not found: " + protocol);
```

Comment:
Please change to 'Protocol not fond' and not add 'protocol'. It's simpler and otherwise the protocol string would have to be escaped as it could otherwise be an potential XSS vuln.